internal/datacoord/index_service.go

```diff
@@ -51,24 +51,25 @@ func (s *Server) startIndexService(ctx context.Context) {
 }
 
 func (s *Server) createIndexForSegment(segment *SegmentInfo, indexID UniqueID) error {
-	indexEngineVersion := Params.CommonCfg.IndexEngineVersion.GetValue()
+	indexEngineVersion := s.IndexEngineVersionManager.GetCurrentIndexEngineVersion()
 	log.Info("create index for segment", zap.Int64("segmentID", segment.ID), zap.Int64("indexID", indexID),
-		zap.String("index_engine_version", indexEngineVersion),
+		zap.Int32("index_engine_version", indexEngineVersion),
```

Comment:
Please also log the minimal version.

Reply:
Meanwhile, do not call the function twice.

Reply:
done